### PR TITLE
Added test for SonataCoreBundle

### PR DESCRIPTION
**Pull request by ahmetakbn**
### Subject

Writing test to increase code coverage of `SonataCoreBundle`

### Todo

- [x] remove @group annotations after WIP
- [x] remove @covers annotations
- [x] line break

### CODE REVIEW

File: Tests/SonataCoreBundleTest.php [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                if ($pass instanceof AdapterCompilerPass) {
+                    return;
+                }
+
+                if ($pass instanceof FormFactoryCompilerPass) {
+                    return;
+                }
+
+                $this->fail(sprintf('Compiler pass is not one of the expected types. Expects "Sonata\AdminBundle\DependencyInjection\Compiler\StatusRendererCompilerPass", "Sonata\AdminBundle\DependencyInjection\Compiler\AdapterCompilerPass" or "Sonata\AdminBundle\DependencyInjection\Compiler\FormFactoryCompilerPass", but got "%s".', get_class($pass)));
+            }));
```
Comment (greg0ire) [inline Tests/SonataCoreBundleTest.php]: You SHOULD use Prophecy (SHOULD as in "the PR will not get refused it you don't, but it's better) ;)

File: Tests/SonataCoreBundleTest.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+                if ($pass instanceof AdapterCompilerPass) {
+                    return;
+                }
+
+                if ($pass instanceof FormFactoryCompilerPass) {
+                    return;
+                }
+
+                $this->fail(sprintf('Compiler pass is not one of the expected types. Expects "Sonata\AdminBundle\DependencyInjection\Compiler\StatusRendererCompilerPass", "Sonata\AdminBundle\DependencyInjection\Compiler\AdapterCompilerPass" or "Sonata\AdminBundle\DependencyInjection\Compiler\FormFactoryCompilerPass", but got "%s".', get_class($pass)));
```
Comment (greg0ire) [inline Tests/SonataCoreBundleTest.php]: Please linebreak this line.

**Comment by greg0ire** [comment]
Excellent initiative @ahmetakbn !

File: Tests/SonataCoreBundleTest.php (hunk 1 above)
Comment (Soullivaneuh) [inline Tests/SonataCoreBundleTest.php]: AFAIK, the discussion is not over: https://github.com/sonata-project/dev-kit/issues/89 :-)

File: Tests/SonataCoreBundleTest.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ * Test for SonataCoreBundle.
+ *
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+final class SonataCoreBundleTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @group core-bundle
+     * @covers Sonata\CoreBundle\SonataCoreBundle::build
```
Comment (Soullivaneuh) [inline Tests/SonataCoreBundleTest.php]: Please remove `@covers`. We want to follow Symfony usage and they don't use it.

File: Tests/SonataCoreBundleTest.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+/**
+ * Test for SonataCoreBundle.
+ *
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+final class SonataCoreBundleTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @group core-bundle
```
Comment (Soullivaneuh) [inline Tests/SonataCoreBundleTest.php]: Why this?

File: Tests/SonataCoreBundleTest.php (hunk 1 above)
Comment (greg0ire) [inline Tests/SonataCoreBundleTest.php]: Oh right, my bad!

File: Tests/SonataCoreBundleTest.php (hunk 4 above)
Comment (ahmetakbn) [inline Tests/SonataCoreBundleTest.php]: This is executing tests separately on my local while working, will be removed.

Comment (OskarStark) [inline Tests/SonataCoreBundleTest.php]: i added a checkbox to todo in the PR header

**Comment by ahmetakbn** [comment]
The only lines not being tested are these in `SonataCoreBundle.php`
```php
if (class_exists('Nelmio\ApiDocBundle\Form\Extension\DescriptionFormTypeExtension')) {
    $formTypes[] = 'nelmio_api_doc.form.extension.description_form_type_extension';
}
```

How should I mock php native functions? any suggestions? May be this: https://github.com/php-mock/php-mock

**Comment by greg0ire** [comment]
I think `NelmioApiDocBundle` should be added to require-dev. You need to rebase to fix StyleCI.

**Comment by ahmetakbn** [comment]
Ok, I will add `NelmioApiDocBundle` to require dev. The StyleCI is failing in other files which are not related to this PR. I will open another PR for that.

**Comment by Soullivaneuh** [comment]
No need for the PR, I'll do that with styleci.

Sullivan SENECHAL

**Comment by greg0ire** [comment]
> Ok, I will add NelmioApiDocBundle to require dev. The StyleCI is failing in other files which are not related to this PR. I will open another PR for that.

Rebasing might be enough for that, I don't know…

**Comment by Soullivaneuh** [comment]
@ahmetakbn I checked StyleCI, nothing to be changed. Be sure your local 3.x branch is sync with upstream and do a rebase again.

**Comment by Soullivaneuh** [comment]
Ouch. I think you did a rebase from master instead of 3.x...

**Comment by greg0ire** [comment]
`git rebase -i origin/3.x` can help you get rid of the unwanted commits

**Comment by ahmetakbn** [comment]
Yes I didn't realize that, sorry. Now it is rebased to 3.x

File: composer.json [hunk 5]
```diff
@@ -37,7 +37,8 @@
         "sensio/framework-extra-bundle": "^2.3 || ^3.0",
         "sonata-project/exporter": "^1.3",
         "symfony/phpunit-bridge": "^2.7",
-        "sllh/php-cs-fixer-styleci-bridge": "^2.0"
+        "sllh/php-cs-fixer-styleci-bridge": "^2.0",
+        "nelmio/api-doc-bundle": "^2.11"
```
Comment (OskarStark) [inline composer.json]: why are we doing this in this PR?

Comment (greg0ire) [inline composer.json]: See previous comment

Comment (OskarStark) [inline composer.json]: got it, sorry

**Comment by ahmetakbn** [comment]
This PR is ready from my side.

File: Tests/SonataCoreBundleTest.php [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+            array('sonata_type_datetime_range_picker', 'Sonata\CoreBundle\Form\Type\DateTimeRangePickerType'),
+            array('sonata_type_equal', 'Sonata\CoreBundle\Form\Type\EqualType'),
+            array('sonata_type_color_selector', 'Sonata\CoreBundle\Form\Type\ColorSelectorType'),
+        );
+    }
+
+    /**
+     * @dataProvider getRegisteredFormMappingAndExtensions
+     *
+     * @param $mapping
```
Comment (Soullivaneuh) [inline Tests/SonataCoreBundleTest.php]: Param type?

File: Tests/SonataCoreBundleTest.php [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+     * @param $extension
```
Comment (Soullivaneuh) [inline Tests/SonataCoreBundleTest.php]: Param type?

**Comment by greg0ire** [comment]
LGTM

**Comment by Soullivaneuh** [comment]
@ahmetakbn I removed the changelog, this is not needed just for test changes.

File: Tests/SonataCoreBundleTest.php (hunk 7 above)
Comment (ahmetakbn) [inline Tests/SonataCoreBundleTest.php]: yes somehow, missing :) will add.

Comment (ahmetakbn) [inline Tests/SonataCoreBundleTest.php]: done

**Comment by Soullivaneuh** [comment]
In the name of the code coverage, I thank you! :+1:

**Comment by greg0ire** [comment]
> +5.4%

Nice!

**Comment by OskarStark** [comment]
Thank you @ahmetakbn kepp the good work 👍

**Pull request merged**